### PR TITLE
Port hat tossing

### PR DESCRIPTION
**Pull request by Linzolle**
## About The Pull Request
The fedora lands neatly on the Captain's head.
Ports #45355 from tg.

## Why It's Good For The Game
It's hilarious, shouldn't be too broken.

## Changelog
:cl:
add: Target head and throw a hat at someone to toss it onto their head, knocking whatever they're wearing off if they are wearing a hat. Some headgear can't be knocked off this way.
/:cl:

### CODE REVIEW

File: code/__DEFINES/obj_flags.dm [hunk 1]
```diff
@@ -2,43 +2,43 @@
 
 
 #define EMAGGED					(1<<0)
-#define IN_USE					(1<<1) // If we have a user using us, this will be set on. We will check if the user has stopped using us, and thus stop updating and LAGGING EVERYTHING!
```
Comment (Putnam3145) [inline code/__DEFINES/obj_flags.dm:5]: this is a whole lot of pointless whitespace changes

Comment (Linzolle) [reply code/__DEFINES/obj_flags.dm:5]: it bothered me that they werent all in line ok

File: code/__DEFINES/obj_flags.dm [hunk 2]
```diff
@@ -48,3 +48,7 @@
 #define ORGAN_EXTERNAL			(1<<3)	//Was this organ implanted/inserted/etc, if true will not be removed during species change.
 #define ORGAN_VITAL				(1<<4)	//Currently only the brain
 #define ORGAN_NO_SPOIL			(1<<5)	//Do not spoil under any circumstances
+
+
+#define SNUG_FIT               	(1<<11) //Prevents knock-off from things like hat-throwing.
+#define ANTI_TINFOIL_MANEUVER   (1<<12) //Hats with negative effects when worn (i.e the tinfoil hat).
```
Comment (Ghommie) [inline code/__DEFINES/obj_flags.dm]: Move these alongside the other clothing flags.

File: code/modules/clothing/head/_head.dm [hunk 3]
```diff
@@ -41,6 +41,49 @@
...
+			if(istype(WH, /obj/item/clothing/head))
+				var/obj/item/clothing/head/WHH = WH
+				///SNUG_FIT hats are immune to being knocked off
+				if(WHH.clothing_flags & SNUG_FIT)
+					H.visible_message("<span class='warning'>[src] bounces off [H]'s [WHH.name]!", "<span class='warning'>[src] bounces off your [WHH.name], falling to the floor.</span>")
+					return
+			///if the hat manages to knock something off
+			if(H.dropItemToGround(WH))
+				H.visible_message("<span class='warning'>[src] knocks [WH] off [H]'s head!</span>", "<span class='warning'>[WH] is suddenly knocked off your head by [src]!</span>")
+		if(H.equip_to_slot_if_possible(src, SLOT_HEAD, 0, 1, 1))
```
Comment (Ghommie) [inline code/modules/clothing/head/_head.dm]:
```suggestion
		if(H.equip_to_slot_if_possible(src, SLOT_HEAD, FALSE, TRUE))
```
The fifth arg is true by default.

**Review by silicons — APPROVED** [review]

**Pull request merged**